### ISSUE TITLE
Discrete (factor) Scales missing Tooltip

### ISSUE THREAD

**Issue by bhive01** [ropensci/plotly#515]
I'm making heatmaps using geom_tile for correlation matrices. 
Using my previous MRE:
```r
require(ggplot2)
require(plotly)

randomName <- function(n) {
    random.string <- rep(NA, n)
    randomizeString <- function(x) {
        a <-sample(letters, 1, replace = TRUE)
        return(a)
    }
    return(paste(c(sapply(random.string, randomizeString, simplify = TRUE)), collapse = ""))
}

xvar = replicate(10, randomName(15))
df1 <- data.frame(x = xvar, y = xvar, z = runif(10) )

g = ggplot(df1, aes(x,y,fill=z)) +
        geom_tile() +
        theme(axis.text.x = element_text(angle = 45, vjust = 1,  size = 12, hjust = 1))
ggplotly(g)
```

When you hover over the volcano examples here: 
https://plot.ly/ggplot2/geom_tile/ 
all of the numbers are lit up in the tooltip. Sometimes these matrices can be large and having it tell you which x and y you're on as well as the value would be really needed. 

This exclusion appears to have been made on purpose:
https://github.com/ropensci/plotly/blob/master/R/ggplotly.R#L195-L203
Though, I'm not sure I understand the code completely. 

When I try and force a tooltip using tooltip = c("y", "x", "fill") it doesn't work. I would really like the ability to force tooltips, because I don't think it's misleading in my case.

**Comment by cpsievert** [comment]
I think I came up with that heuristic before I implemented `tooltip`...since it's now easy to hide tooltip info, I think it does make sense to show discrete positional scales in the tooltip by default.

**Comment by cpsievert** [comment]
After thinking about this some more, it only makes sense to provide the _domain_ value in the tooltip (i.e., the value before [`ggplot2:::map_position()`](https://github.com/ropensci/plotly/blob/908cbf178bf5e48662a7efa2a2a0159b529157b9/R/ggplotly.R#L76) is used to transform the positions). I'm pretty sure this is do-able provided statistic(s) do not reduce the dimension of the scale.

There's also the question of what should happen when custom breaks/labels are provided in `scale_x/y_discrete()`, but I'm not sure the time required to implement would be worth it.

**Comment by cpsievert** [comment]
tada! http://cpsievert.github.io/plotly-test-table/a90d523/heatmap-discrete-tooltip/

**Comment by cpsievert** [comment]
Fixed via 346d964

**Comment by bhive01** [comment]
Beautiful! Thanks. 

![screen shot 2016-03-17 at 7 23 12 am](https://cloud.githubusercontent.com/assets/394973/13848964/53383802-ec11-11e5-92f5-8d156e2119f8.png)
